uberwriter/markup_regex.py

```diff
@@ -21,9 +21,9 @@
 BLOCK_QUOTE = re.compile(
     r"^ {0,3}(?:> ?)+(?P<text>.+)", re.M)
 HEADER = re.compile(
-    r"^ {0,3}(?P<level>#{1,6}) (?P<text>[^\n]+)", re.M)
+    r"^ {0,3}(?P<level>#{1,6})(?P<text>[^\n]+)", re.M)
 HEADER_UNDER = re.compile(
-    r"(?:^\n*|\n\n)(?P<text>[^\s].+)\n {0,3}[=\-]+(?:\s+?\n|$)")
+    r"(?:^\n*|\n\n)(?P<text>[^\s].+)\n[=\-]+(?: +?\n|$)")
```

Comment:
The switch from `\s` to ` ` makes perfect sense, but the removal of the 3 optional spaces in the beginning is incorrect, at least in Commonmark and GitHub-Flavored Markdown which are arguably the most commonly used implementations. :blush:

There is a bug though: the 3 optional spaces are also allowed in the first line, not only the second one.